### PR TITLE
Prepend NGSI-v1 to the v1 tests, NGSI-v2 to the v2 tests

### PR DESCRIPTION
**Pull request by jason-fox**
Simple Grep rename of the descriptions of the **NGSI v2** tests, to distinguish them from the **NGSI-LD** test suite

Also updated the **NGSI-v1** test files which have equivalent tests to distinguish between v1, v2 and LD

### CODE REVIEW

**Comment by jason-fox** [comment]
I'm not sure of your PR philosophy here, given that this PR touches tests only should I:

a) Make minimal changes
b) Prettify the changed test files - to reduce the scope of #753 
c) Update the tests to run using ES6 syntax - to reduce the scope of #831

File: test/unit/ngsiv2/expressions/expressionBasedTransformations-test.js [hunk 1]
```diff
@@ -193,7 +193,7 @@ var iotAgentLib = require('../../../../lib/fiware-iotagent-lib'),
         providerUrl: 'http://smartGondor.com'
     };
 
-describe('Expression-based transformations plugin', function() {
+describe('NGSI-v2 - Expression-based transformations plugin', function() {
```
Comment (fgalan) [inline test/unit/ngsiv2/expressions/expressionBasedTransformations-test.js]: Not sure but many of these tests are "parallel" to existing ones in NGSIv1 (maybe @dcalvoalonso could confirm, as he did great part of the implementation).

In that case, maybe it's a good idea to add a `'NGSI-v1 - ` prefix also in those ones, for the sake of completeness.

**Comment by fgalan** [comment]
> I'm not sure of your PR philosophy here, given that this PR touches tests only should I:
> 
> a) Make minimal changes (as is)
> b) Prettify the changed test files - to reduce the scope of #753
> c) Update the tests to run using ES6 syntax - to reduce the scope of #831

This case is different from https://github.com/telefonicaid/iotagent-node-lib/pull/841#issuecomment-584621453 in the sense in this case the test files are not new but pre-existing.

In my opinion, the decision should depend on the impact on other people's ongoing works. In other words, are the test files you plan to change already being changed in some of the ongoing PRs we have in the repository (apart from yours and mine)?

**Comment by jason-fox** [comment]
Current list of amended tests in outstanding PRs.

```
test/unit/ngsiv2/provisioning/device-provisioning-api_test.js 
test/unit/ngsiv2/provisioning/updateProvisionedDevices-test.js
test/unit/provisioning/device-group-api-test.js

test/unit/expressions/expressionCombinedTransformations-test.js
test/unit/expressions/jexlBasedTransformations-test.js
test/unit/expressions/jexlExpression-test.js
test/unit/ngsiv2/expressions/expressionCombinedTransformations-test.js
test/unit/ngsiv2/expressions/jexlBasedTransformations-test.js

test/unit/general/iotam-autoregistration-test.js
test/unit/provisioning/device-group-utils_test.js
test/unit/provisioning/singleConfigurationMode-test.js
```

File: test/unit/ngsiv2/expressions/expressionBasedTransformations-test.js (hunk 1 above)
Comment (jason-fox) [reply test/unit/ngsiv2/expressions/expressionBasedTransformations-test.js]: Fixed  718ca29

**Comment by jason-fox** [comment]
> are the test files you plan to change already being changed in some of the ongoing PRs 

Since the update now touches both NGSI-v1 and NGSI-v2 the answer would be *YES*
I'd prefer just to complete the title changes for now.

**Pull request merged**